Parallelize Antora component build

gpx1000 opened the pull request.

gpx1000 committed:
Revert "Update dependencies and fix web manifest path"

This reverts commit 7daf96271cbf3425550da8f950f0c32a2f68e117.

gpx1000 committed:
Update @antora/lunr-extension to use Khronos Group repository

gpx1000 commented:
Yeah I made a mistake so reverted it..  it still needs the change merged

On Sat, Aug 30, 2025, 7:01 PM Jon Leech ***@***.***> wrote:

> *oddhack* left a comment (KhronosGroup/Vulkan-Site#131)
> <https://github.com/KhronosGroup/Vulkan-Site/pull/131#issuecomment-3239677633>
>
> This is odd - it is open but looks like it has already been merged to
> main. Did you push this directly to main branch as well as open the MR?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/KhronosGroup/Vulkan-Site/pull/131#issuecomment-3239677633>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA5IAY3PIIBDG4VXSFRYKUT3QJJONAVCNFSM6AAAAACFGN7QUSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTEMZZGY3TONRTGM>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

oddhack commented:
> Yeah I made a mistake so reverted it.. it still needs the change merged

Are you saying you want this branch merged as it is, or that you want to do something to this branch before we merge it? Confused.

gpx1000 commented:
No, now that you guys have seen it the UI branch with the fix needs to be
merged so this will also need the new changes to point to the main instead
of the branch then it needs to be merged.   That's why I left it as it is
now.

On Sun, Aug 31, 2025, 1:00 PM Jon Leech ***@***.***> wrote:

> *oddhack* left a comment (KhronosGroup/Vulkan-Site#131)
> <https://github.com/KhronosGroup/Vulkan-Site/pull/131#issuecomment-3240375925>
>
> Yeah I made a mistake so reverted it.. it still needs the change merged
>
> Are you saying you want this branch merged as it is, or that you want to
> do something to this branch before we merge it? Confused.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/KhronosGroup/Vulkan-Site/pull/131#issuecomment-3240375925>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA5IAY2PBX4A3S7LIM2XTLD3QNH6RAVCNFSM6AAAAACFGN7QUSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTENBQGM3TKOJSGU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

gpx1000 committed:
Add sources-parallel Antora extension scaffold

Introduce initial scaffold for the sources-parallel extension to enable per-source/module parallel execution in Antora. Updated playbook, scripts, and Makefile to integrate the extension, while keeping it safe/inert for future parallelism iterations.

gpx1000 committed:
Introduce `antora-sources-parallel` extension

Added a new Antora extension `antora-sources-parallel` to enable parallel-friendly builds. Includes concurrency hints (Phase 1) and an experimental per-source fan-out orchestrator for parallel execution. Added documentation, sample configurations, and supporting scripts for integration.

gpx1000 committed:
Add `antora:ci` script and optimize CI caching for parallel builds

Enhanced the Antora parallel sources extension with a fast path for CI builds (`--no-lunr`) and updated documentation with performance tips. Improved CI workflows with better caching strategies for Antora artifacts to boost efficiency.

gpx1000 committed:
Enhance `antora-sources-parallel` with path rebasing, Lunr optimization, and global index build

Added utilities to rebase relative playbook paths for proper resolution. Improved shard builds by removing Lunr extension per shard and introducing a single global Lunr index pass. Ensured global search-index artifacts are correctly copied to the final site.

gpx1000 committed:
Expand `antora-sources-parallel` to shard sources by `start_paths` for finer-grained parallelism.

gpx1000 committed:
Add configurable Antora cache directory and optimize Lunr indexing

Introduced the ability to set a custom `ANTORA_CACHE_DIR` and ensured directories are created as needed. Improved Lunr global index builds with a fast-path option and fallback to legacy Antora mode for robustness. Enhanced logging for cache usage and indexing steps.

gpx1000 committed:
Add global Lunr index builder to `antora-sources-parallel` extension

Introduced a new script `build-lunr-from-site.js` to generate a global search index directly from the merged Antora site. Updated `package.json` to include the `lunr` dependency required for indexing.

gpx1000 committed:
Optimize Lunr indexing with conservative IO concurrency and memory limits

Refactored Lunr builder to use minimal memory by limiting IO concurrency through `ANTORA_LUNR_IO_WORKERS` (default: 2) and capping characters per page (`ANTORA_LUNR_MAX_CHARS`, default: 200,000). Replaced custom parallel pool logic with a streamlined approach and added elapsed time logging for indexing.

gpx1000 committed:
Further reduce memory and CPU usage during Lunr indexing

Set stricter defaults for IO concurrency (`ANTORA_LUNR_IO_WORKERS=1`) and max characters per page (`ANTORA_LUNR_MAX_CHARS=60,000`). Added a simplified pipeline (`ANTORA_LUNR_SIMPLE_PIPELINE=1`) to minimize resource use by skipping heavy stemming and stopword filters. Restricted indexing to `latest` pages when applicable.

gpx1000 committed:
Increase V8 heap for Lunr indexing during CI builds

Added `ANTORA_LUNR_NODE_MAX_OLD_SPACE` to allow configurable heap size (`--max-old-space-size`) for Lunr index serialization. Updated CI workflow and `fanout.js` to set conservative default values for memory and pipeline options.

oddhack commented on Makefile:
```diff
@@ -79,22 +79,22 @@ prep-docs:
 # of the "Edit this Page" link otherwise generated.
 export CI = true
-build-site:
+build-site: build-ui prep-sources
```
I don't want building the site to clean and regenerate everything in the component repositories every time, which is what this will do AFAICT.

oddhack reviewed (changes requested):
Per [my other comment](https://github.com/KhronosGroup/Vulkan-Site/pull/131#discussion_r2383247658) I don't want the Makefile target to completely re-do the markup transformation work every time, so holding approval back until that's addressed. Otherwise I can't really speak to the parallelization but if it's faster than what we're doing now and returns the same output, I'll be happy with it.

oddhack commented:
Telecon: requested @gpx1000 give this a more useful PR title reflecting its main purpose in parallel builds. Will be reverted to Draft status until some further changes are made.

oddhack commented:
I updated the title because the previous title was not informative and I think this matches the purpose of the PR.
